specs/SIM/alert_manager/SODA_AlertManagerDesign.md

```diff
@@ -0,0 +1,239 @@
+|              | Input: alert source configuration |
+|              | Output: Success/Failure |
+| API server   | Name: HandleClearAlert |
+|              | Description: To process clear alert |
+|              | Input: clear alert related data |
+|              | Output: Success/Failure |
+
+### Alert API model
+
+![](Alert_APIs.png)
+
+Swagger Reference (https://github.com/sodafoundation/delfin/blob/master/openapi-spec/swagger.yaml)
+
+### Data Model
+* AlertSourceInfo: All the information about the alert source device which will be stored in DB
+
+![](Alert_Source_Data_Model.png)
+
+* AlertModel: Soda alert model which will be pushed to export manager after complete processing of trap
+
+| Attribute      | Data Type    | Unit         | Enumerated Value                         | Constraint Level | Description                              |
+|----------------|--------------|--------------|------------------------------------------|------------------|------------------------------------------|
+| category       | Enum(int)    |              |                                          | Mandatory        | Type of the reported alarm notification  |
+|                |              |              | 1:New                                    |                  |                                          |
+|                |              |              | 2:Fixed                                  |                  |                                          |
+|                |              |              | 3:Removed                                |                  |                                          |
+|                |              |              | 4:Recovered                              |                  |                                          |
+|                |              |              | 5:Alarm acknowledgement change           |                  |                                          |
+|                |              |              | 6:Alarm unacknowledgement                |                  |                                          |
+|                |              |              | 7:Other change                           |                  |                                          |
+| occurUtcTime   | long         | milliseconds |                                          | Mandatory        | UTC time when the alarm is generated (in milliseconds) |
+| matchKey       | String(255)  |              |                                          | Mandatory        | Alarm matching key                       |
+| meDn           | String(255)  |              |                                          | Not              | Unique Id of alarm source device in the whole management system. |
+| meName         | String(255)  |              |                                          | Mandatory        | Alarm source which identifies the device that generates an alarm. Alarm source can be identified by the device name and IP address. |
```

Comment:
is this name final "meName" "MeDn" etc..

Reply:
Udpated in latest commit